Documentation/sourcebuild-in-repos/new-dependencies.md

```diff
@@ -0,0 +1,46 @@
+# Adding a new dependency
+
+## Basics
+When adding a new dependency, there are a few steps common between any type of dependency.  If you use darc, `darc add-dependency` will take care of this process for you.  Otherwise:
+1. Add the package and a default version to eng/Versions.props.  The default version property should be named `<PackageName>PackageVersion`, e.g. `SystemCollectionsImmutablePackageVersion`.
+2. Add the package and a repo reference to eng/Version.Details.xml.  This should include a SourceBuild metadata entry, looking something like:
+   ```<Dependency Name="Microsoft.Net.Compilers.Toolset" Version="4.2.0-3.22205.5" CoherentParentDependency="Microsoft.NET.Sdk">
+      <Uri>https://github.com/dotnet/roslyn</Uri>
+      <Sha>0167599e0e1634ea3ed8d0e41390a3c0d9b3e4e9</Sha>
+      <SourceBuild RepoName="roslyn" ManagedOnly="true" />
+    </Dependency>```
+3. Set up dependency flow from the foreign repo to your repo.
+
+In general, you should aim to use one version of each package.  If you are using a package as reference-only, it is possible to use multiple versions, but only one implementation version of each package will be used - source-build will override it to the version that is being built in this version of the SDK.
+
+The source-build metadata is important - this tells source-build which repo package contains the specific nupkg you want.  A `<SourceBuild>` tag means that the dependency is needed both in the individual repo builds and the overall tarball build.  A `<SourceBuildTarball>` tag can be used for temporary issues when the required package is not included in the source-build intermediate package for that repo, or the source-build intermediate package is not produced.
+
+Another uncommon case that can cause problems is when the repo `<Dependency>` version and the source-built intermediate package version don't match.  In this case, a direct dependency should be added to the source-build intermediate package and the CoherentParentDependency attribute should be set to the repo that consumes the dependency. For an example, see [installer's F# dependency](https://github.com/dotnet/installer/blob/ba1739a2363b1062f03ea386ec67174c6468d3b2/eng/Version.Details.xml#L128).  You can find the version needed by running `darc get-build` or using [BAR](https://aka.ms/bar).
+
+## Microsoft dependencies
+
+.NET dependencies are from a Microsoft .NET project.  Adding a new .NET dependency is usually quite easy, especially if the repo uses Arcade to build.  Internal dependencies are from Microsoft repos but not .NET.  Adding a new internal dependency is usually pretty straightforward but does have some edge cases.  Use this checklist:
```

Comment:
I am not following the following two sentences

> .NET dependencies are from a Microsoft .NET project. 
 
> Internal dependencies are from Microsoft repos but not .NET.

Please help me understand.

Reply:
Added some more elaboration and examples.